test/skate_web/controllers/detours_controller_test.exs

```diff
@@ -407,6 +414,76 @@ defmodule SkateWeb.DetoursControllerTest do
       )
     end
 
+    @tag :authenticated
+    test "sends data to TripModificationPublisher", %{conn: conn} do
+      route_pattern = build(:gtfs_route_pattern)
+      shape_with_stops = build(:shape_with_stops)
+
+      connection_start = Util.Location.as_location!(build(:gtfs_shape_point))
+      connection_end = Util.Location.as_location!(build(:gtfs_shape_point))
+      missed_stop = Enum.at(shape_with_stops.stops, 1)
+
+      reassign_env(:skate_web, :route_pattern_fn, fn _ -> route_pattern end)
+      reassign_env(:skate_web, :shape_with_stops_fn, fn _ -> shape_with_stops end)
+
+      reassign_env(:skate_web, :missed_stops_fn, fn _ ->
+        build(:missed_stops_result,
+          missed_stops: [missed_stop]
+        )
+      end)
+
+      missed_stop_id = missed_stop.id
+      representative_trip_id = route_pattern.representative_trip_id
+
+      expect(
+        Skate.Detours.MockTripModificationPublisher,
+        :publish_modification,
+        fn trip_modification, shape, opts ->
+          assert %TripModification{
+                   modifications: [
+                     %TripModification.Modification{
+                       start_stop_selector: %TripModification.StopSelector{
+                         stop_id: ^missed_stop_id
+                       },
+                       end_stop_selector: %TripModification.StopSelector{stop_id: ^missed_stop_id}
+                     }
+                   ],
+                   selected_trips: [
+                     %TripModification.SelectedTrip{
+                       trip_ids: [^representative_trip_id],
+                       shape_id: shape_id
+                     }
+                   ]
+                 } = trip_modification
+
+          assert %Shape{
+                   shape_id: ^shape_id
+                 } = shape
+
+          assert opts == [is_draft?: true]
+        end
+      )
+
+      post(conn, ~p"/api/detours/finished_detour",
```

Comment:
There's no verification after this post that the response indicated success, and I gather that's unneeded?

Reply:
There are other tests that validate that the response was successful. This one is solely concerned with the `TripModification` and `Shape` messages.

Reply:
> What is the purpose of the post, then?

The `post` is what triggers the call to the mocked module which asserts that the function is called with the right parameters. Without this post, `mox` helpfully fails the test[^mox-failure] because `publish_modification` is never called, therefore the assertions in the defined in the `expect` are never run.

[^mox-failure]: 
	```
	  1) test finished_detour/2 sends data to TripModificationPublisher (SkateWeb.DetoursControllerTest)
	     test/skate_web/controllers/detours_controller_test.exs:418
	     ** (Mox.VerificationError) error while verifying mocks for #PID<0.551.0>:
	
	       * expected Skate.Detours.MockTripModificationPublisher.publish_modification/3 to be invoked once but it was invoked 0 times
	```